client/state/document/reducer.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+import filter from 'lodash/filter';
+
+/**
+ * Internal dependencies
+ */
+import {
+	DOCUMENT_LINK_ADD,
+	DOCUMENT_LINK_REMOVE,
+	DOCUMENT_META_ADD,
+	DOCUMENT_META_REMOVE,
+	DOCUMENT_DESCRIPTION_SET,
+	DOCUMENT_TITLE_SET,
+	DOCUMENT_UNREAD_COUNT_SET
+} from 'state/action-types';
+
+export function title( state = '', action ) {
+	switch ( action.type ) {
+		case DOCUMENT_TITLE_SET:
+			return action.title;
+	}
+
+	return state;
+}
+
+export function unread_count( state = 0, action ) {
+	switch ( action.type ) {
+		case DOCUMENT_UNREAD_COUNT_SET:
+			return action.count;
+	}
+
+	return state;
+}
+
+export function description( state = '', action ) {
+	switch ( action.type ) {
+		case DOCUMENT_DESCRIPTION_SET:
+			return action.description;
+	}
+
+	return state;
+}
+
+export function meta( state = [], action ) {
+	switch ( action.type ) {
+		case DOCUMENT_META_ADD:
+			return [...state, action.meta];
+
+		case DOCUMENT_META_REMOVE:
+			return filter( state, ( value, index ) => index !== action.index );
```

Comment:
Removing by `index` will be difficult in practice, as we won't know what the index is without manually iterating over all `meta` in state. It'd be nice if there were an obvious property to key by, but both `meta` and `link` are pretty varied in their usage (`name`, `property`, `charset` for `meta`, multiple `rel` for link).

We should consider:

1. Will we ever need to remove `meta` or `link`?
2. If so, perhaps the action creator could accept a "search object", by which the reducer could omit via [`_.reject`](https://lodash.com/docs#reject).

Example:

```js
reject( [
	{ rel: 'canonical', href: 'http://mysite.com/example' },
	{ rel: 'apple-touch-icon', href: 'http://mysite.com/img/apple-touch-icon-57x57.png' },
	{ rel: 'apple-touch-icon', sizes: '72x72', 'href': 'http://mysite.com/img/apple-touch-icon-72x72.png' }
], { rel: 'apple-touch-icon', sizes: '72x72' } )
```